spot_rl_experiments/spot_rl/envs/place_env.py

```diff
@@ -2,69 +2,216 @@
+
+        self.place_env = SpotPlaceEnv(config, spot)
+        self.place_env.power_robot()
+
+    def execute(self, place_target_list, is_local=False):
+        """
+        Execute place for each place target in place_target_list
+
+        Args:
+            place_target_list (list): List of place targets to go and place
+            is_local (bool): Whether the place target is in the local frame of the robot
+
+        Returns:
+            success_list (list): List of dicts containing the following keys:
+                - time_taken (float): Time taken to place the object
+                - success (bool): Whether the place was successful
+                - place_target (np.array([x,y,z])): Place target in base frame
+                - ee_pos (np.array([x,y,z])): End effector position in base frame
+        """
+        success_list = []
+        for place_target in place_target_list:
+            start_time = time.time()
+            if self.use_policies:
+                observations = self.place_env.reset(place_target, is_local)
+                done = False
+
+                while not done:
+                    action = self.policy.act(observations)
+                    observations, _, done, _ = self.place_env.step(arm_action=action)
+
+            else:
+                # Get reset arm position (Is there a better way to do this without using place_env?????????????)
+                self.place_env.reset(place_target, is_local)
+
+                # End effector positions in base frame (as needed by the API)
+                curr_ee_pos = self.place_env.get_gripper_position_in_base_frame()
+                goal_ee_pos = self.place_env.get_corrected_place_target_in_base_frame()
+                intr_ee_pos = generate_intermediate_point(curr_ee_pos, goal_ee_pos)
+
+                # Get direction vector from current ee position to goal ee position for EE orientation
+                dir_rpy_to_goal = get_RPY_from_vector(goal_ee_pos - curr_ee_pos)
+
+                # Go to intermediate point
+                cmd_id = self.spot.move_gripper_to_point(
+                    intr_ee_pos,
+                    dir_rpy_to_goal,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+                # Direct the gripper to face downwards
+                dir_rpy_to_goal = [0.0, np.pi / 2, 0.0]
+
+                # Go to goal point
+                cmd_id = self.spot.move_gripper_to_point(
+                    goal_ee_pos,
+                    dir_rpy_to_goal,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+            success_list.append(
+                {
+                    "time_taken": time.time() - start_time,
+                    "success": self.place_env.place_attempted or not self.use_policies,
```

Comment:
@akshararai should we keep the "success" not dependent on the environment?

But instead make the use of this [function ](https://github.com/facebookresearch/spot-sim2real/pull/65/files#diff-e7e2ddf17a2682546b2138af6f61a30f6932934396051f244a6d640a57c4c33bR142-R167) ??

Reply:
well this is not checking success at all right now right?

Reply:
`place_attempted` will be true (from `base_env`) iff robot reaches the desired (x,y,z) target.